measurement/plugins/netflix_fast/measurements.py

```diff
@@ -0,0 +1,334 @@
+        except ConnectionError as e:
+            return self._get_netflix_error("netflix-script-response", traceback=str(e))
+
+        try:
+            token = re.search(r'token:"(.*?)"', script_resp.text).group(1)
+        except AttributeError:
+            return self._get_netflix_error(
+                "netflix-token-regex", traceback=script_resp.text
+            )
+
+        try:
+            self._query_api(s, token)
+        except ConnectionError as e:
+            return self._get_netflix_error("netflix-api-response", traceback=str(e))
+        except json.decoder.JSONDecodeError as e:
+            return self._get_netflix_error("netflix-api-json", traceback=str(e))
+        except TypeError as e:
+            return self._get_netflix_error("netflix-api-parse", traceback=str(e))
+        except KeyError as e:
+            return self._get_netflix_error("netflix-api-parse", traceback=str(e))
+
+        try:
+            conns = [
+                self._get_connection(target["url"]) for target in self.thread_results
+            ]
+        except ConnectionError as e:
+            return self._get_netflix_error("netflix-connection", traceback=str(e))
+
+        fast_data = self._manage_threads(conns)
+
+        return NetflixFastMeasurementResult(
+            id=self.id,
+            download_rate=float(fast_data["speed_mbps"]),
+            download_rate_unit=NetworkUnit("Mbit/s"),
+            download_size=float(fast_data["total"]),
+            download_size_unit=StorageUnit("B"),
+            asn=self.client_data["asn"],
+            ip=self.client_data["ip"],
+            isp=self.client_data["isp"],
+            city=self.client_data["location"]["city"],
+            country=self.client_data["location"]["country"],
+            urlcount=self.urlcount,
+            reason_terminated=fast_data["reason_terminated"],
+            errors=[],
+        )
+
+    def _manage_threads(self, conns):
+        # Create worker threads
+        threads = [None] * len(self.thread_results)
+        for i in range(len(self.thread_results)):
+            threads[i] = Thread(
+                target=self._threaded_download,
+                args=(conns[i], self.thread_results[i], time.time()),
+            )
+            threads[i].daemon = True
+            threads[i].start()
+
+        start_time = time.time()
+        recent_measurements = deque(maxlen=STABLE_MEASUREMENTS_LENGTH)
+        percent_deltas = deque(maxlen=STABLE_MEASUREMENTS_LENGTH)
+        while True:
+            elapsed_time = time.time() - start_time
+            total = 0
+            for thread_result in self.thread_results:
+                total += thread_result["download_size"]
+            speed_kBps = total / elapsed_time / 1024
+            recent_measurements.append(speed_kBps)
+
+            if len(recent_measurements) == 10:
+                # Calculate percentage difference to the average of the last ten measurements
+                percent_deltas.append(
```

Comment:
Was outlier detection required here?

Reply:
Probably not required to begin with.

But a short comment noting its absence would probably be a good addition.

Reply:
@jamesaj23 as per comments above we're not checking for outliers here. That's ok but it'd be good to add a comment somewhere acknowledging that. Perhaps in the docstring for `_is_stabilised`?

> Determines whether measurements results are stabilised by... blah blah.
>
> Note that there is currently no accounting for outliers. All measurements are regarded equally.

Looking at this has also made me realise: do we not need to limit `percent_deltas` to _"the last ten measurements"_? Right now I _think_ we're determining if the measurement `_is_stabilised` based on the `max` of _all_ the deltas for the lifetime of the measurement (including those that happened before the last 10).